src/main/java/book/store/onlinebookstore/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package book.store.onlinebookstore.service.impl;
+
+import book.store.onlinebookstore.dto.userdto.UserRegistrationRequestDto;
+import book.store.onlinebookstore.dto.userdto.UserResponseDto;
+import book.store.onlinebookstore.exceptions.RegistrationException;
+import book.store.onlinebookstore.mapper.UserMapper;
+import book.store.onlinebookstore.model.User;
+import book.store.onlinebookstore.repository.UserRepository;
+import book.store.onlinebookstore.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+
+    @Override
+    public UserResponseDto authenticate(UserRegistrationRequestDto userRegistrationRequestDto)
+            throws RegistrationException {
+        if (userRepository.findByEmail(userRegistrationRequestDto.getEmail()) == null) {
```

Comment:
use ifPresent() to check user's existing and throw exception 
```suggestion
        userRepository.findByEmail(userEmail)
            .ifPresent(existingUser -> {
                throw new RegistrationException("User by email " + userEmail + " already exists");
            });
```

Reply:
it s doesnt work(

Reply:
![Uploading Screenshot_9.jpg…]()

Reply:
в таких випадках, коли саджещн від ментора не працює, пінгуй в чаті

Reply:
ok thx